arangod/Indexes/Index.h

```diff
@@ -481,6 +485,58 @@ struct AttributeAccessParts {
   aql::AstNodeType opType;
 };
 
+struct VectorIndexRandomVector {
+  double bParam;
+  double wParam;
+  std::vector<double> v_param;
+
+  template<class Inspector>
+  friend inline auto inspect(Inspector& f, VectorIndexRandomVector& x) {
+    return f.object(x).fields(f.field("bParam", x.bParam),
+                              f.field("wParam", x.wParam),
+                              f.field("vParam", x.v_param));
+  }
+};
+
+struct VectorIndexDefinition {
+  std::size_t dimensions;
+  double min;
+  double max;
+  std::size_t Kparameter;
+  std::size_t Lparameter;
+  std::vector<VectorIndexRandomVector> randomFunctions;
+
+  template<class Inspector>
+  friend inline auto inspect(Inspector& f, VectorIndexDefinition& x) {
+    return f.object(x)
+        .fields(f.field("dimensions", x.dimensions), f.field("min", x.min),
+                f.field("max", x.max), f.field("Kparameter", x.Kparameter),
+                f.field("Lparameter", x.Lparameter),
+                f.field("randomFunctions", x.randomFunctions))
+        .invariant([](VectorIndexDefinition& x) -> inspection::Status {
+          if (x.dimensions < 1) {
+            return {"Dimensions must be greated then 0!"};
```

Comment:
Also for all following occurrences:
```suggestion
            return {"Dimensions must be greater than 0!"};
```